test/sys/test_wait.rs

```diff
@@ -37,6 +37,19 @@ fn test_wait_exit() {
     }
 }
 
+#[test]
+fn test_waitstatus_pid() {
+    let _m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
+
+    match fork().unwrap() {
+        Child => {},
```

Comment:
Yeah, the forked process duplicates the entire test harness.  But it doesn't duplicate any threads other than the one that called `fork`.  So if one of those other threads held a mutex that is required by this thread during its teardown phase, then this thread will deadlock.  The reason to use `_exit` instead of `sys::process::exit` is subtler, and it's because `_exit` doesn't do any C-level teardown.  See [fork(2)](http://man7.org/linux/man-pages/man2/fork.2.html).